### ISSUE TITLE
AMQP event source backoff config seems to be required

### ISSUE THREAD

**Issue by yuriylsh** [argoproj/argo-events#340]
**Describe the bug**
We have a setup for amqp (RabbitMq) event source/gateway/sensor. The configs are basically taken from https://github.com/argoproj/argo-events/tree/master/examples. When we use event source example that include backoff, everything works. As soon as we remove the backoff config, we are getting the following error in the gateway (from `kubectl -n argo-events logs pod/postlogservice-gateway -c amqp-events`):

```
starting gateway server
INFO[2019-08-15 14:53:53] operating on event source                     event-source=rabbitmq-with-retry
goroutine 50 [running]:
runtime/debug.Stack(0x31, 0xc0003b1ae0, 0x2)
        /usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/argoproj/argo-events/gateways.Recover(0xc0004b60a0, 0x13)
        /home/vaibhav/go/src/github.com/argoproj/argo-events/gateways/gateway.go:53 +0x10c
panic(0x12a03e0, 0x2140510)
        /usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/argoproj/argo-events/gateways/core/stream/amqp.(*AMQPEventSourceExecutor).listenEvents(0xc0004be698, 0xc00029c140, 0xc00032e060, 0xc0003480c0, 0xc000348120, 0xc000348180)
        /home/vaibhav/go/src/github.com/argoproj/argo-events/gateways/core/stream/amqp/start.go:74 +0xbb
created by github.com/argoproj/argo-events/gateways/core/stream/amqp.(*AMQPEventSourceExecutor).StartEventSource
        /home/vaibhav/go/src/github.com/argoproj/argo-events/gateways/core/stream/amqp/start.go:42 +0x21e
```

**To Reproduce**
Follow the amqp example for event source, gateway, sensor (https://github.com/argoproj/argo-events/tree/master/examples). Delete the "example-with-retry" from `data` section from https://github.com/argoproj/argo-events/blob/master/examples/event-sources/amqp.yaml. Deploy to kubernetes and gateway will fail with the error above. Put "example-with-retry" back and remove "example-without-retry" entry. Re-deploy. Everything works as expected
**Expected behavior**
The following event source config (taken from examples) should not cause gateway to fail:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: amqp-event-source
  labels:
    # do not remove
    argo-events-event-source-version: v0.10
data:
  # no retries if connection to amqp service is not successful
  example-without-retry: |-
    # url of the service
    url: amqp://amqp.argo-events:5672
    # name of the exchange
    exchangeName: "name of the exchange"
    # type of the exchange
    exchangeType: fanout
    # routing key for the exchange
    routingKey: "routing key"
```

**Environment (please complete the following information):**
 - argo-events-0.4.2

**Comment by VaibhavPage** [comment]
I'll take a look.

**Comment by VaibhavPage** [comment]
fixed in v0.12
